### PR TITLE
Add the ability to set custom converters to avro compiler

### LINKED ISSUE #21529: quarkus-avro: support UUID logical type
### Description

Currently, the "uuid" logical type in the .avsc schema is interpreted as a String, and there's no validation to check if this string is a valid UUID.

### Implementation ideas

We can use an existing `org.apache.avro.Conversions.UUIDConversion` class and register it on the compiler.

### PR DESCRIPTION
**Pull request by glefloch**
* This add the `UUIDConversion` converter to the avro compiler
* This allows to set custom conversion classes to the avro compiler

closes #21529

### CODE REVIEW

File: extensions/avro/deployment/src/main/java/io/quarkus/avro/deployment/AvroCodeGenProviderBase.java [hunk 1]
```diff
@@ -153,6 +161,8 @@ public boolean trigger(CodeGenContext context) throws CodeGenException {
             gettersReturnOptional = getBooleanProperty("avro.codegen.gettersReturnOptional", false);
             optionalGettersForNullableFieldsOnly = getBooleanProperty("avro.codegen.optionalGettersForNullableFieldsOnly",
                     false);
+            String conversions = prop("avro.codegen.customConversions", "");
+            this.customConversions = "".equals(conversions) ? EMPTY : conversions.split(",");
```
Comment (cescoffier) [inline extensions/avro/deployment/src/main/java/io/quarkus/avro/deployment/AvroCodeGenProviderBase.java:165]: We should trim the values.

**Review by cescoffier — CHANGES_REQUESTED** [review]
Looks great! I just added a very small comment.

File: extensions/avro/deployment/src/main/java/io/quarkus/avro/deployment/AvroCodeGenProviderBase.java (hunk 1 above)
Comment (glefloch) [reply extensions/avro/deployment/src/main/java/io/quarkus/avro/deployment/AvroCodeGenProviderBase.java]: Sure, done :)

**Review by cescoffier — APPROVED** [review]

**Comment by glefloch** [comment]
@michalszynkiewicz could you have a look ?

**Review by michalszynkiewicz — APPROVED** [review]

**Pull request merged**